node/bindnode/repr.go

```diff
@@ -320,7 +322,7 @@ _skipAbsent:
 	if err != nil {
 		return 0, nil, err
 	}
-	if value.IsAbsent() {
+	if w.nextIndex > w.reprEnd {
```

Comment:
Is it possible to be in a situation where `value.IsAbsent()` and we're not at the end amongst trailing absents? e.g. `[absent,present,present,absent,absent]`? Is there a use of this iterator where that can happen, perhaps by constructing this from the typed form and then list iterating on the repr form?

Reply:
Absents can only be at the end, as per the undocumented spec :) See https://github.com/ipld/go-ipld-prime/blob/e39d20bf18dcc21a50ef2005bae6de11e5c311e0/schema/gen/go/genStructReprTuple.go#L13.

Arguably this needs to be properly documented in the specs, which is why I'll raise an issue elsewhere. But in short: we should support trailingn optionals in bindnode, to at least match codegen in terms of features.